### DISCUSSION TITLE
Integrated Gcode converter

### DISCUSSION THREAD

**Discussion by faisalbelwadi1997** [winder/Universal-G-Code-Sender#1668] [Help]
I have a setup where i need to print alphanumeric characters on jobs one after another. ex. ABC123 , ABC124 etc. ( I am using a 2axis CNC)
It is difficult to create a gcode everytime and load it and then send it through UGS for each iteration.

My aim is to have a text to gcode converter like StickFont integrated in UGS which would take text as an input, generate gcode and send it to the machine. For the next iteration, i just have to increment the counter and the loop continues.
Can it be done via UGS plugin?
I am not a software developer. Can anyone please point me in the right direction as to how to integrate these for my purpose.
Any other suggestions are also welcome

**Comment by breiler** [comment]
This is a pretty cool use case but is unfortunately not something I have done before. 

StickFont seems to come with command line options which would make it plausible to use for generating gcode in an automated process. I have not looked at the file format, but if you open a file up in notepad you can try changing the text from there. If that works you could create a template file where you just "search and replaced" within the file to generate your different files.
Check out its manual http://ncplot.com/stickfontv2/StickFont_Manual.pdf. We will probably not make a plugin for it as StickFont is closed source and Windows only.

I am trying to add a text tool to the designer tool in UGS (#1634) which could maybe solve this for you?

**Reply by faisalbelwadi1997** [reply]
Thanks for the reply Joacim.
The text tool in designer would probably do the job and would be really great. 
Are you planning to use single stroke fonts for it?
Is it possible to add a functionality which could add headers for the gcode
I will be eagerly waiting for the next build
Thanks again for the amazing work :)

**Reply by breiler** [reply]
The fonts are loaded from your operating system. If you want single stroke fonts you can install them to your system and they should become available in UGS.
